simtools/simtel/simtel_io_histogram.py

```diff
@@ -248,44 +265,50 @@ def total_area(self):
             Total area covered on the ground covered by the simulation.
         """
         if self._total_area is None:
+            events_histogram, _ = self.fill_event_histogram_dicts()
+            self._initialize_histogram_axes(events_histogram)
 
             if self.area_from_distribution is True:
-                events_histogram, _ = self.fill_event_histogram_dicts()
-                self._initialize_histogram_axes(events_histogram)
                 area_from_distribution_max_radius = 1.5 * np.average(
                     self.radius_axis[:-1], weights=np.sum(events_histogram["data"], axis=0)
                 )
                 self._total_area = (np.pi * (area_from_distribution_max_radius * u.m) ** 2).to(
                     u.cm**2
                 )
             else:
+                # The max of the core range is always half the upper edge:
+                # self.radius_axis[-1]/2 is equal to self.config["core_range"][1]
                 self._total_area = (
-                    np.pi
-                    * (
-                        ((self.config["core_range"][1] - self.config["core_range"][0]) * u.m).to(
-                            u.cm
-                        )
-                    )
-                    ** 2
+                    np.pi * (((self.radius_axis[-1] / 2 - self.radius_axis[0]) * u.m).to(u.cm)) ** 2
                 )
         return self._total_area
 
-    @property
-    def energy_range(self):
+    def _set_energy_range(self):
         """
         Energy range used in the simulation.
+        If the input file is a .simtel.zst, the energy range is taken from its configuration,
+        written in the file itself.
+        If the input file is a .hdata.zst, the energy range has to be passed as argument to the
+        class. An error is raised in case the argument is missing.
 
-        Returns
-        -------
-        list:
-            Energy range used in the simulation [min, max]
-        """
-        if self._energy_range is None:
-            self._energy_range = [
-                self.config["E_range"][0] * u.TeV,
-                self.config["E_range"][1] * u.TeV,
-            ]
-        return self._energy_range
+        Raises
+        ------
+        ValueError:
+            if input parameter is missing.
+        """
+        if self.energy_range is None:
+            try:
+                self.energy_range = [
+                    self.config["E_range"][0] * u.TeV,
+                    self.config["E_range"][1] * u.TeV,
+                ]
+            except TypeError as exc:  # E_range not in self.config
+                msg = "energy_range needs to be passed as argument (a list of energies in TeV)."
```

Comment:
Same here, 'list of energies' is not what we want - we want min/max of the energy range.

Reply:
Done.